### PR TITLE
Add RFC about relations in our developer portal catalog

### PR DESCRIPTION
**Pull request by marians**
Towards https://github.com/giantswarm/giantswarm/issues/27854

The content intro should explain the purpose of this PR quite well.

### CODE REVIEW

**Comment by marians** [comment]
FYI @gusevda This is by far not ready for review yet. Just letting you know it's here.

File: devportal-component-relationships/README.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+In a Go project, the fact that a module depends on another module is encoded in the `go.sum` file. Alternatively, we can also investigate the [Github API for exorting a bill of materials](https://docs.github.com/en/rest/dependency-graph/sboms?apiVersion=2022-11-28).
+
+Other languages we use (Python, JavaScript/Typescript, ...) have similar concepts and may be explored, however each of those is only a niche in our landscape and we will have to evaluate whether the value gained is worth the effort.
+
+## Components depending on other components (services)
+
+We have cases where a certain service can only function when another service is also deployed with it. Examples:
+
+- `kubectl-gs` (CLI) depends on `athena` (service) running in whatever cluster to sign in to.
```
Comment (uvegla) [inline devportal-component-relationships/README.md:58]: This is only true for parts of `kubectl-gs` actually, `login` I assume. You could say the same for `template` to depend on `Vault` or `SOPS` key being present, but those also depend on installation.

File: devportal-component-relationships/README.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+- `passage` depends on `userd`.
+
+TODO: Find some more modern examples
+
+While some of these dependencies might be useful to depict, it's also possible to take this too far. For example, every app (as in the Giant Swarm app platform) relies on app-operator and chart-operator to deliver it. Howevwer, rendering this dependency for hundreds of apps is likely not helping anyone. So we need human judgement when deciding which dependencies to depict.
+
+## App collections and the apps they include
+
+TODO Can we detect an app collection?
```
Comment (uvegla) [inline devportal-component-relationships/README.md:72]: There is a naming convention in place, I think there is even an ADR / PDR somewhere about it but I cannot seem to find it. :/

File: devportal-component-relationships/README.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+TODO Can we find the apps belonging to a collection in a robust way? [Example](https://github.com/giantswarm/capa-app-collection/blob/2e34e0580e50418897d1b169771e3d92aadf4eb1/flux-manifests/athena.yaml)
+
+## App bundles and the apps they provide
+
+App bundles should point to the apps they provide, using the `hasParts` relation. In the opposite direction, this also means that from an app, there would be relation to an app bundle (or, potentially, several app bundles) including this app.
+
+For example, the `security-bundle` should point to `kyverno`, `falco`, `starboard-exporter` and so on using the `hasParts` relation.
```
Comment (uvegla) [inline devportal-component-relationships/README.md:80]: Can we parse this from the repository to have it up-to-date all the time?

File: devportal-component-relationships/README.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+The data needed is already encoded in the [bundle values](https://github.com/giantswarm/security-bundle/blob/14eb70be3cf65f4b78c046a0bc3fa5ccca72c565/helm/security-bundle/values.yaml).
```
Comment (uvegla) [inline devportal-component-relationships/README.md:82]: That makes things very complicated, but what about versions? Bundles have version and different versions include and depend on different apps.

File: devportal-component-relationships/README.md (hunk 1 above)
Comment (marians) [reply devportal-component-relationships/README.md:58]: Yes, this example is for the `login` command. Adding clarification.

File: devportal-component-relationships/README.md (hunk 4 above)
Comment (marians) [reply devportal-component-relationships/README.md:82]: In my thinking, the devportal should rather represent the latest state (default branch) than give access to the "historical" details. Just to keep complexity down.

**Pull request closed without merging**